src/Core/ImageWorkshopLayer.php

```diff
@@ -913,8 +919,11 @@ public function resizeByNarrowSideInPercent($newNarrowSideWidth, $conversePropor
      * @param int $newNarrowSideWidth
      * @param boolean $converseProportion
      */
-    public function resizeByNarrowSide($unit = self::UNIT_PIXEL, $newNarrowSideWidth, $converseProportion = false)
+    public function resizeByNarrowSide($unit, $newNarrowSideWidth, $converseProportion = false)
     {
+        if (!in_array($unit, [self::UNIT_PIXEL,self::UNIT_PERCENT])) {
+            $unit = self::UNIT_PIXEL;
```

Comment:
```suggestion
            throw new \InvalidArgumentException("Incorrect unit '$unit' parameter.");
```

Reply:
Ok got it ! the goal was to keep the idea of the default value.
In PHP8, all args with default value should be at the end and mandatory first.

Reply:
Yes I understand.
But, it's a mistake in the actual code base :)

If the parameter is invalid, we shouldn't continue executing the code because we don't know what to do.
The developer should fix his code.